ddtrace/tracer/option.go

```diff
@@ -186,16 +186,27 @@ func WithPropagator(p Propagator) StartOption {
 	}
 }
 
-// WithServiceName sets the default service name to be used with the tracer. It is
-// deprecated in favour of WithService and will be removed in the next major version.
+// WithServiceName is deprecated. Please use WithService.
+// If you are using an older version and you are upgrading from WithServiceName
+// to WithService, please note that WithService will determine the service name of
+// server and framework integrations. This provides a better experience of the
+// product.
```

Comment:
nit: we can probably leave out the last sentence.

Reply:
👍